File: front/src/schema.ts
interface Category {
  id: number;
  name: string;
}

interface Achievement {
  id: number;
  name: string;
  description: string;
  points: number;
  categoryId: number;
}

interface User {
  id: number;
  username: string;
}

interface UserExperience {
  userId: number;
  categoryId: number;
  experience: number;
}

interface UserLevel {
  userId: number;
  categoryId: number;
  level: number;
}

interface UserAchievement {
  userId: number;
  achievementId: number;
}

interface Task {
  id: number;
  userId: number;
  name: string;
  description: string;
  points: number;
  categoryId: number;
}

interface SubTask {
  id: number;
  taskId: number;
  name: string;
  description: string;
  points: number;
}

File: schema.py
from dataclasses import dataclass, field

@dataclass
class Category:
    id: int
    name: str

@dataclass
class Achievement:
    id: int
    name: str
    description: str
    points: int
    categoryId: int

@dataclass
class User:
    id: int
    username: str

@dataclass
class UserExperience:
    userId: int
    categoryId: int
    experience: int

@dataclass
class UserLevel:
    userId: int
    categoryId: int
    level: int

@dataclass
class UserAchievement:
    userId: int
    achievementId: int

@dataclass
class Task:
    id: int
    userId: int
    name: str
    description: str
    points: int
    categoryId: int

@dataclass
class SubTask:
    id: int
    taskId: int
    name: str
    description: str
    points: int
